packages/react-core/src/components/Select/Select.tsx

```diff
@@ -16,10 +35,16 @@ export interface SelectProps extends MenuProps, OUIAProps {
   /** Renderer for a custom select toggle. Forwards a ref to the toggle. */
   toggle: (toggleRef: React.RefObject<any>) => React.ReactNode;
   /** Function callback when user selects an option. */
-  onSelect?: (event?: React.MouseEvent<Element, MouseEvent>, itemId?: string | number) => void;
+  onSelect?: (
+    event?: React.MouseEvent<Element, MouseEvent>,
+    itemId?: string | number,
+    toggleRef?: React.RefObject<any>
+  ) => void;
   /** Callback to allow the select component to change the open state of the menu.
-   * Triggered by clicking outside of the menu, or by pressing either tab or escape. */
+   * Triggered by clicking outside of the menu, or by pressing either tab or escape (or specificed in onOpenChangeKeys). */
   onOpenChange?: (isOpen: boolean) => void;
+  /** Keys that trigger onOpenChange, defaults to tab and escape. */
+  onOpenChangeKeys?: string[];
```

Comment:
The comment I had left in the Menu PR would apply here and possibly require updating (depending where we land). https://github.com/patternfly/patternfly-react/pull/9014#discussion_r1184094156

Reply:
Replied in the other PR, for others let's try to keep the conversation in one PR and I can update based on the results of that.